Replace deprecated Litle URLs

grantbdev opened the pull request:
This PR [replaces deprecated Litle URLs with the new Vantiv URLs](https://www.vantiv.com/content/dam/vantiv/merchants-partners/docs/ecommerce-domains-faq_v1-apr-2017.pdf).

grantbdev committed:
Vantiv: Replace deprecated Litle URLs

- Replace deprecate Litle URLs with the updated Vantiv URLs.

grantbdev committed:
Vantiv: Update certification tests

- Update expectations to match the newer certification test responses
  so that the certification tests pass.

grantbdev commented:
@tconst Updated this PR to follow the newly rebased `master` and `add-vantiv-gateway` and also update the remote certification tests.

grantbdev committed:
Vantiv: Make exp date year dynamic in tests

- Make the credit card expiration date dynamic
  based on the credit card object year
  so the tests continue to pass as the credit card test helper
  automatically moves up the expiration year based on the current year.

tconst commented on lib/active_merchant/billing/gateways/vantiv.rb:
```diff
@@ -10,7 +10,7 @@ module Billing #:nodoc:
     # name.
     class VantivGateway < Gateway
       self.test_url = "https://www.testlitle.com/sandbox/communicator/online"
```
Is there an updated test url?

grantbdev replied:
There was not at that time, but there is [now](https://developer.vantiv.com/message/3721). Updating.

grantbdev committed:
Vantiv: Unfreeze request headers

- Call `#dup` before returning our request headers
  because the `DEFAULT_HEADERS` constant is a frozen hash
  and `dup` copies the hash in an unfrozen state.
  This is necessary due to an update in ActiveMerchant connections
  that mutate the headers hash thus raising an exception
  and failing the connection.
  https://github.com/activemerchant/active_merchant/pull/2868

grantbdev committed:
Vantiv: Remove remote sandbox tests

- Remove the remote tests for the Vantiv sandbox
  because they have been neglected and no longer pass
  since the gateway has been modified without updating these tests.
  They can be removed because the unit tests
  and remote certification tests cover the same functionality
  and the prelive environment is more useful than sandbox.
- Change the default test URL to the prelive URL.

grantbdev commented:
@tconst Rebased `add-vantiv-gateway` on latest upstream master and then updated this PR with the bare minimum to keep the tests passing.
